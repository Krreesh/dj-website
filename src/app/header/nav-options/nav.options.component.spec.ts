import {ComponentFixture, TestBed} from '@angular/core/testing';
import {NavOptionsComponent} from './nav.options.component';
import {RouterTestingModule} from '@angular/router/testing';
import {TranslatePipeMock} from '../../translation/translate.pipe.mock';

describe('NavOptionsComponent', () => {
  let component: NavOptionsComponent;
  let fixture: ComponentFixture<NavOptionsComponent>;

  beforeEach(() => {
    TestBed.configureTestingModule({
      imports: [RouterTestingModule],
      declarations: [NavOptionsComponent, TranslatePipeMock]
    });
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(NavOptionsComponent);
    component = fixture.debugElement.componentInstance;
  });

  it('should emit event when navigated to a navigation option', () => {
    const navigationOption = `website.vision`;

    const spy = jest.spyOn(component.navigated, 'emit');
    fixture.detectChanges();

    findNavigationOption(navigationOption).click();

    expect(spy).toHaveBeenCalled();
  });

  function findNavigationOption(navigationOption: string) {
    return fixture.nativeElement.querySelector(
      `[data-navigation-option="${navigationOption}"]`
    );
  }
});
