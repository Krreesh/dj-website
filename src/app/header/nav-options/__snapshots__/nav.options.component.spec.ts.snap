// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavOptionsComponent should render 1`] = `
<app-nav-options
  navigated={[Function EventEmitter]}
  navigationOptions={[Function Array]}
  router={[Function Router]}
>
  <ul
    class="navbar-nav"
  >
    
    
    <li
      class="nav-item active"
    >
      <a
        class="nav-link navigation"
        data-navigation-option="website.home"
        href="/"
      >
        website.home.translated
      </a>
    </li>
    
    <li
      class="nav-item"
    >
      <a
        class="nav-link navigation"
        data-navigation-option="website.vision"
        href="/vision"
      >
        website.vision.translated
      </a>
    </li>
    
    <li
      class="nav-item"
    >
      <a
        class="nav-link navigation"
        data-navigation-option="website.cv"
        href="/cv"
      >
        website.cv.translated
      </a>
    </li>
    
    <li
      class="nav-item"
    >
      <a
        class="nav-link navigation"
        data-navigation-option="website.projects"
        href="/projects"
      >
        website.projects.translated
      </a>
    </li>
    
    <li
      class="nav-item"
    >
      <a
        class="nav-link navigation"
        data-navigation-option="website.blogposts"
        href="/blogposts"
      >
        website.blogposts.translated
      </a>
    </li>
    
    <li
      class="nav-item"
    >
      <a
        class="nav-link navigation"
        data-navigation-option="website.data"
        href="/data"
      >
        website.data.translated
      </a>
    </li>
  </ul>
</app-nav-options>
`;
